Automattic/wp-calypso
# LogRocket: Enable network logging

Resolves https://github.com/Automattic/jetpack-backup-team/issues/485

> [!IMPORTANT]
> LogRocket is currently disabled in production. The purpose of this PR is to be ready for when we decide to enable it.

## Proposed Changes

* Enable [LogRocket's network logging](https://docs.logrocket.com/reference/network#disable-recording-of-network-data) on Jetpack Cloud, excluding any request and response body and authentication header.

## Why are these changes being made?


* To explore the full potential of LogRocket trial, we need to capture network logs. For privacy reasons, we are sanitizing any request and response body.

## Testing Instructions
* Spin up a Jetpack Cloud Live branch and append `?flags=ad-tracking,logrocket` to the URL to enable the feature flags.
* Enable debug logs using the developer console:
  `localStorage.setItem( 'debug', 'calypso:*' );`
* Filter console and network tabs by `logrocket`
* Navigate to the `Backup` or `Settings` page and ensure you see these debug records:
   ![CleanShot 2024-04-08 at 00 04 11@2x](https://github.com/Automattic/wp-calypso/assets/1488641/45cb2bfc-64a6-4029-8a13-83fdab7f88e9)
* Check on `Network` tab that there is API calls to LogRocket including a request payload similar to this:
  ![CleanShot 2024-05-20 at 19 15 31@2x](https://github.com/Automattic/wp-calypso/assets/1488641/3697adf5-99d1-441a-8f53-eb5fb2a9eff2)


## Pre-merge Checklist
